e2e/fixtures/use-router/src/TestRouter.tsx

```diff
@@ -0,0 +1,37 @@
+'use client';
+
+import { Link, useRouter_UNSTABLE } from 'waku';
+
+export default function TestRouter() {
+  const router = useRouter_UNSTABLE();
+  // TODO: Why is this necessary? On the client `searchParams`
+  //       is a URLSearchParams object, but on the server it's an array.
```

Comment:
I didn't know it's an array on the server. How can it be??

Reply:
Not sure, but without that, it breaks with `get is not a function`. And the debugger showed me an array. Maybe node version differences?

Reply:
Can you dig into it further? If it's a bug in Waku, we would like to fix it.

Reply:
Sorry, took me some time to get back to this. I was not able to pinpoint whats the problem. This is what I have found.

At this point, `searchParams` is still an `URLSearchParams` object:
https://github.com/dai-shi/waku/blob/f61b264a466bec88c3c3bfcc5588a83bd39ea4c9/packages/waku/src/router/define-router.ts#L217-L228

But when `ServerRouter` gets rendered, it has become `[]`:
https://github.com/dai-shi/waku/blob/f61b264a466bec88c3c3bfcc5588a83bd39ea4c9/packages/waku/src/router/client.ts#L530-L552

My guess is that the route definition gets serialized by react when passed into the `ServerRouter` client component and the empty `URLSearchParams` map gets turned into `[]`?

With https://github.com/dai-shi/waku/pull/746/commits/ddb40868b4bb75f29218aab781ba5643fb0ec68b I removed the additional construction of `URLSearchParams` to see if it also happens on CI, which it does.

Reply:
@dai-shi : i tried to find out more, but basically thats it. An empty `URLSearchParams` object gets passed as a prop into `createElement` and inside the `ServerRouter` its an empty array. It has to be reacts prop-handling. Maybe it would be safer to stringify `searchParams`?

Reply:
Yeah, you are absolutely right.

repro:
```bash
$ npm create waku@next
$ cd ...
$ npm i
$ cat foo.cjs
const { renderToPipeableStream } = require('react-server-dom-webpack/server');

const data = new URLSearchParams();

renderToPipeableStream(data).pipe(process.stdout);
$ node --conditions react-server foo.cjs
0:[]
```

I will open another PR to make it a string and see how it goes.

Reply:
#799